File: website/typegraphs/oauth2.ts
// skip:start
import { Policy, t, typegraph } from "@typegraph/sdk/index.js";
import { Auth } from "@typegraph/sdk/params.js";
import { DenoRuntime } from "@typegraph/sdk/runtimes/deno.js";

// skip:end

typegraph({
  name: "oauth2-authentication",
}, (g) => {
  const deno = new DenoRuntime();
  const pub = Policy.public();

  const ctx = t.struct({ "exp": t.integer().optional().fromContext("exp") });

  // highlight-start
  g.auth(
    Auth.oauth2Github("openid profile email"),
  );
  // highlight-end

  g.expose({
    get_context: deno.identity(ctx),
  }, pub);
});

File: website/typegraphs/oauth2.py
# skip:start
from typegraph import typegraph, Policy, t, Graph
from typegraph.graph.params import Auth, Cors
from typegraph.runtimes.deno import DenoRuntime

# skip:end


@typegraph(
    # skip:next-line
    cors=Cors(allow_origin=["https://metatype.dev", "http://localhost:3000"]),
)
def oauth2_authentication(g: Graph):
    deno = DenoRuntime()
    public = Policy.public()

    ctx = t.struct({"exp": t.integer().optional().from_context("exp")})

    # highlight-start
    g.auth(Auth.oauth2_github("openid profile email"))
    # highlight-end

    g.expose(
        public,
        get_context=deno.identity(ctx),
    )
